dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/messaging/SingleDataRequestEmailMessageSender.kt

```diff
@@ -0,0 +1,104 @@
+package org.dataland.datalandcommunitymanager.services.messaging
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.dataland.datalandbackend.openApiClient.api.CompanyDataControllerApi
+import org.dataland.datalandbackend.openApiClient.model.DataTypeEnum
+import org.dataland.datalandmessagequeueutils.cloudevents.CloudEventMessageHandler
+import org.dataland.datalandmessagequeueutils.constants.ExchangeName
+import org.dataland.datalandmessagequeueutils.constants.MessageType
+import org.dataland.datalandmessagequeueutils.constants.RoutingKeyNames
+import org.dataland.datalandmessagequeueutils.messages.InternalEmailMessage
+import org.dataland.datalandmessagequeueutils.messages.TemplateEmailMessage
+import org.dataland.keycloakAdapter.auth.DatalandJwtAuthentication
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import java.util.*
+
+/**
+ * A class that manages generating emails messages for bulk and single data requests
+ */
+@Component
+class SingleDataRequestEmailMessageSender(
+    @Autowired private val cloudEventMessageHandler: CloudEventMessageHandler,
+    @Autowired private val objectMapper: ObjectMapper,
+    @Autowired val companyApi: CompanyDataControllerApi,
+) : DataRequestEmailMessageSenderBase() {
+    private val logger = LoggerFactory.getLogger(SingleDataRequestEmailMessageSender::class.java)
+
+    /**
+     * Function that generates the message object for single data request mails
+     */
+    fun sendSingleDataRequestInternalMessage(
+        userAuthentication: DatalandJwtAuthentication,
+        datalandCompanyId: String,
+        dataType: DataTypeEnum,
+        reportingPeriods: Set<String>,
+    ) {
+        val correlationId = UUID.randomUUID().toString()
+        val companyName = companyApi.getCompanyInfo(datalandCompanyId).companyName
+        logger.info(
+            "User with Id ${userAuthentication.userId} has submitted a single data request for company with" +
+                " Id $datalandCompanyId and correlationId $correlationId",
+        )
```

Comment:
Auch hier würde ich (ähnlich wie beim bulk data request) sagen dass das Loggen der Infos über die submission eher in die processSingleDataRequest Funktion gehört.
Die Funktion hier sollte sich ganz allein um das Schreiben der Message kümmern.
CompanyName und CorrelationId könnte man ja als Funktionsinputs reingeben.

Reply:
Am Telefon besprochen.

Reply:
Erledigt.